okhttp/src/test/java/okhttp3/OpenJSSETest.kt

```diff
@@ -0,0 +1,67 @@
+package okhttp3
+
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import okhttp3.tls.HandshakeCertificates
+import okhttp3.tls.HeldCertificate
+import okhttp3.tls.internal.TlsUtil.localhost
+import org.junit.After
+import org.junit.Assert
+import org.junit.Assert.*
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.openjsse.net.ssl.OpenJSSE
+import java.net.InetAddress
+import java.security.Security
+
+class OpenJSSETest {
+  @JvmField @Rule val clientTestRule = OkHttpClientTestRule()
+  @JvmField @Rule val server = MockWebServer()
+  lateinit var client: OkHttpClient
+
+  @Before
+  fun setUp() {
+    Security.insertProviderAt(OpenJSSE(), 1)
```

Comment:
We may want to use junit 5 https://junit.org/junit5/docs/current/api/org/junit/jupiter/api/parallel/ResourceLock.html

To avoid weird interactions via providers?

Reply:
JUnit 5 is a big can of worms because they broke API compatibility and MockWebServer needs JUnit 4.

Reply:
Do you have a plan for how that will work? https://github.com/square/okhttp/issues/4667 ?

It would be good to break that dependency and have it working with both.  Over time more users will expect Junit 5.